File: homework_6/view/ParseLine.java
package view;
import java.util.*;

//класс для парсинга введенного пользователем в терминале примера вычислений типа "23.8 + 85.7"
//используемые операторы +, -, *, /
//пробелы между числом и оператором обязательны!
public class ParseLine {
    private String line;
    private String[] parseLine;

    //считывание строки
    public ParseLine() {
        System.out.println("Enter number1, operation (+, -, *, /), number2 with a space: ");
        Scanner in = new Scanner(System.in);
        if (in.hasNextLine()) {
            line = in.nextLine();
            parseLine = line.split(" ");
        }
        in.close();
    }

    //первое число
    public double getA() {
        return Double.valueOf(parseLine[0]);
    }

    //второе число
    public double getB() {
        return Double.valueOf(parseLine[2]);
    }

    //оператор
    public String getOperation() {
        return parseLine[1];
    }

}

File: homework_7/view/parse_line.py
"""класс для парсинга введенного пользователем в терминале примера вычислений типа "23.8 + 85.7"
используемые операторы +, -, *, /
пробелы между числом и оператором обязательны!"""

def parse_line():
    example = input("Enter calculation example: ")
    list_ex = list(example.split(' '))
    return list_ex

def get_a(list_ex):
    a = float(list_ex[0])
    return a

def get_b(list_ex):
    b = float(list_ex[2])
    return b

def get_operation(list_ex):
    return list_ex[1]
